plenum/test/test_metrics_stats.py

```diff
@@ -0,0 +1,177 @@
+from datetime import datetime
+
+import pytest
+import statistics
+
+from plenum.common.metrics_collector import KvStoreMetricsCollector, MetricsType
+from plenum.common.metrics_stats import ValueAccumulator, load_metrics_from_kv_store, MetricsStats
+from plenum.test.test_metrics_collector import generate_events, MockTimestamp
+from storage.kv_store import KeyValueStorage
+from storage.kv_store_leveldb import KeyValueStorageLeveldb
+from storage.kv_store_rocksdb import KeyValueStorageRocksdb
+
+db_no = 0
+
+
+@pytest.yield_fixture(params=['rocksdb', 'leveldb'])
+def storage(request, tdir) -> KeyValueStorage:
+    global db_no
+    if request.param == 'leveldb':
+        db = KeyValueStorageLeveldb(tdir, 'metrics_ldb_{}'.format(db_no))
+    else:
+        db = KeyValueStorageRocksdb(tdir, 'metrics_rdb_{}'.format(db_no))
+    db_no += 1
+    yield db
+    db.close()
+
+
+def test_value_accumulator_dont_return_anything_when_created():
+    acc = ValueAccumulator()
+    assert acc.count == 0
+    assert acc.sum == 0
+    assert acc.avg is None
+    assert acc.stddev is None
+    assert acc.min is None
+    assert acc.max is None
+
+
+def test_value_accumulator_can_add_value():
+    value = 4.2
+    acc = ValueAccumulator()
+    acc.add(value)
+    assert acc.count == 1
+    assert acc.sum == value
+    assert acc.avg == value
+    assert acc.stddev is None
+    assert acc.min == value
+    assert acc.max == value
+
+
+def test_value_accumulator_handles_same_values():
+    value = 4.2
+    count = 5
+    acc = ValueAccumulator()
+    for _ in range(count):
+        acc.add(value)
+
+    assert acc.count == count
+    assert acc.sum == value * count
+    assert acc.avg == value
+    assert acc.stddev == 0
+    assert acc.min == value
+    assert acc.max == value
+
+
+def test_value_accumulator_can_add_several_values():
+    values = [4.2, -1.3, 10.8]
+    acc = ValueAccumulator()
+    for value in values:
+        acc.add(value)
+
+    assert acc.count == len(values)
+    assert acc.sum == sum(values)
+    assert acc.avg == statistics.mean(values)
```

Comment:
Why do we use `statistics` module only in tests and not in `ValueAccumulator` for calculation?

Reply:
@ashcherbakov because ValueAccumulator doesn't store values (there can be billions of them when analyzing metrics from a long run), and one of purposes of this test is to be sure that it's implementation is accurate enough